### ISSUE TITLE
Error in delta usersync: Error in API call: Invalid property: roles

### ISSUE THREAD

**Issue by LRSFC-DanJ** [microsoft/o365-moodle#1566]
We have just upgraded our Moodle to version 3.9.4 and installed the latest 3.9 compatible plugin versions of the Microsoft 365 plugin set, and we have followed the instructions at https://docs.microsoft.com/en-us/microsoftteams/platform/resources/moodleinstructions but are getting stuck on Step 13 where it says to test the Sync to Azure AD. If we force a full sync this works, but without this option selected we are receiving:

```
Sync users with Azure AD.
Execute scheduled task: Sync users with Azure AD. (local_o365\task\usersync)
... started 16:04:46. Current memory use 12.9MB.
...... Starting sync
...... No skiptoken (delta) stored.
...... No deltatoken stored.
...... Using delta sync.
...... Contacting Azure AD...
...... Error in delta usersync: Error in API call: Invalid property: roles
...... Resetting skip and delta tokens.
...... Got response from Azure AD
...... Clearing deltatoken (none received)
...... Clearing skiptoken (delta) (none received)
...... No users received to sync.
...... Sync process finished.
... used 5 dbqueries
... used 0.34180903434753 seconds
Scheduled task complete: Sync users with Azure AD. (local_o365\task\usersync)
```

Can anyone suggest what the issue may be? We did the permission checks so we think we have all the correct permissions set for the Azure AD app side.

**Comment by paoloviglione** [comment]
The same behaviour with:

Moodle 3.10.1
Microsoft 365 Integration 3.10.0

(full sync works)

**Comment by pete1854** [comment]
I have the same issue. Moodle 3.10
Microsoft 365 Integration 3.10.0

**Comment by ETS-MB** [comment]
Same issue here : 

Moodle      :  3.9.4 (Build: 20210122)
local_o365 :  3.9.3   2020071505

Execute scheduled task: Synchronisez les utilisateurs avec Azure AD. (local_o365\task\usersync)
...... Starting sync
...... No skiptoken (delta) stored.
...... No deltatoken stored.
...... Using delta sync.
...... Contacting Azure AD...
...... Error in delta usersync: Erreur lors de l'appel d'API : Invalid property: roles
...... Resetting skip and delta tokens.
...... Got response from Azure AD
...... Clearing deltatoken (none received)
...... Clearing skiptoken (delta) (none received)
...... No users received to sync.
...... Sync process finished.
... used 12 dbqueries
... used 0.71537590026855 seconds

**Comment by thebenkahn** [comment]
I have the same behavior on a site after upgrading from 3.7 to 3.10. I hope the PR can be integrated soon :-)

**Comment by weilai-irl** [comment]
Hi @thebenkahn,

The fix will be included in the March 2021 release, which should come out before the end of March. At the mean time, feel free to apply the patch according to the PR - the change is quite simple and affect only one line of code.

Regards,
Lai

**Comment by weilai-irl** [comment]
Hi all,

The fix to this issue has been included in the latest 3.9.4 and 3.10.1 release. Please upgrade the plugins and try again.

I'll close the issue for now. Please feel free to re-open the issue, or create a new one, if the issue still happens to you.

Regards,
Lai
